### PR TITLE
Bump pymemcache version

### PR DESCRIPTION
**Pull request by dumpstate**
Following pymemcache release notes: https://github.com/pinterest/pymemcache/releases/tag/v3.0.0.

### CODE REVIEW

File: microcosm_caching/memcached.py [hunk 1]
```diff
@@ -69,15 +63,13 @@ def __init__(
...
-        deserializer=json_deserializer,
+        serde=None,
         testing=False,
     ):
         client_kwargs = dict(
             connect_timeout=connect_timeout,
             timeout=read_timeout,
-            serializer=serializer,
-            deserializer=deserializer,
+            serde=serde if serde is not None else JsonSerde(),
```
Comment (simpleigh-globality) [inline microcosm_caching/memcached.py]: Nit: seems weird to have a `not` condition here in what's basically a ternary. What about:

```suggestion
            serde=serde or JsonSerde(),
```

File: microcosm_caching/memcached.py [hunk 2]
```diff
@@ -25,36 +25,30 @@ class SerializationFlag(IntEnum):
     JSON = 2
 
 
-def json_serializer(key, value):
+class JsonSerde:
```
Comment (simpleigh-globality) [inline microcosm_caching/memcached.py]: Nit: I know this is the parameter name, but it still seems a bit cryptic. Maybe `JsonSerializer`, or `JsonSerializerDeserializer`, or just expand the docstring to use the term "deserializer"?

File: microcosm_caching/memcached.py (hunk 1 above)
Comment (simpleigh-globality) [inline microcosm_caching/memcached.py]: Is it worth adding tests that `MemcachedCache` can:

1. use a sensible default value
1. use a different serializer if one is provided

File: microcosm_caching/memcached.py (hunk 2 above)
Comment (dumpstate) [reply microcosm_caching/memcached.py]: I was following `pymemcache` convention. But definitely the name is a little cryptic.

**Pull request merged**